### PR TITLE
SDT-79 Create Scheduled Task for Purge Stored Proc

### PR DESCRIPTION
**Pull request by markdathornehmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SDT-79

### Change description ###

Create Scheduled Task for Purge Stored Proc

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: services/src/unit-test/java/uk/gov/moj/sdt/services/PurgeServiceTest.java
Comment (paulridout) [inline services/src/unit-test/java/uk/gov/moj/sdt/services/PurgeServiceTest.java:1]: To increase the coverage, add some tests for when the commitIntervalString is an empty string, not an empty string etc.  This may require the addition of a setCommitInterval() method to the PurgeService class.

**Review by paulridout — CHANGES_REQUESTED** [review]

File: dao/src/integ-test/java/uk/gov/moj/sdt/dao/PurgeNativeCallFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        checkTestDataBeforeRunningPurge();
+
+        purgeNativeCallFunction.executePurgeStoredProc(1000);
+
+        checkTestDataAfterRunningPurge();
+
+        LOGGER.info("TEST COMPLETED: the Purge has been run");
+    }
+
+    private void  checkTestDataBeforeRunningPurge() {
```
Comment (paulridout) [inline dao/src/integ-test/java/uk/gov/moj/sdt/dao/PurgeNativeCallFunctionTest.java:82]: There's a double space between "void" and the method name.  Same for checkTestDataAfterRunningPurge().

File: dao/src/integ-test/resources/uk/gov/moj/sdt/dao/sql/PurgeDaoTest.sql [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  VALUES (11, 1, 4, 3, 'BR103', 'CR104', (now() - INTERVAL '100 day'), 1, 'BSSTATUS1', now(), now(), 'EC104', 'This is error 1', 1, 'BLK PAYLOAD 1');
+INSERT INTO bulk_submissions(
+  bulk_submission_id, bulk_customer_id, target_application_id, service_request_id, sdt_bulk_reference, customer_reference, created_date, number_of_requests, bulk_submission_status, completed_date, updated_date, error_code, error_text, version_number, bulk_payload)
+  VALUES (12, 1, 4, 3, 'BR104', 'CR105', (now() - INTERVAL '60 day'), 1, 'BSSTATUS2', now(), now(), 'EC105', 'This is error 2', 1, 'BLK PAYLOAD 2');
+INSERT INTO bulk_submissions(
+  bulk_submission_id, bulk_customer_id, target_application_id, service_request_id, sdt_bulk_reference, customer_reference, created_date, number_of_requests, bulk_submission_status, completed_date, updated_date, error_code, error_text, version_number, bulk_payload)
+  VALUES (13, 1, 4, 3, 'BR105', 'CR106', (now() - INTERVAL '30 day'), 1, 'BSSTATUS3', now(), now(), 'EC106', 'This is error 3', 1, 'BLK PAYLOAD 3');
+
+
```
Comment (paulridout) [inline dao/src/integ-test/resources/uk/gov/moj/sdt/dao/sql/PurgeDaoTest.sql:31]: Any reason these two sets of bulk_submissions test data are separated from each other by a double blank line?  If not, remove the blank lines.

File: dao/src/main/java/uk/gov/moj/sdt/dao/ErrorLogDao.java
Comment (paulridout) [inline dao/src/main/java/uk/gov/moj/sdt/dao/ErrorLogDao.java:1]: May not need the comment block at the beginning of the class as it applies to the previous development process.  If it is needed then change the copyright year to the current year.

File: dao/src/main/java/uk/gov/moj/sdt/dao/PurgeJobAuditDao.java
Comment (paulridout) [inline dao/src/main/java/uk/gov/moj/sdt/dao/PurgeJobAuditDao.java:1]: May not need the comment block at the beginning of the class as it applies to the previous development process. If it is needed then change the copyright year to the current year.

File: dao/src/main/java/uk/gov/moj/sdt/dao/PurgeJobAuditMessageDao.java
Comment (paulridout) [inline dao/src/main/java/uk/gov/moj/sdt/dao/PurgeJobAuditMessageDao.java:1]: May not need the comment block at the beginning of the class as it applies to the previous development process. If it is needed then change the copyright year to the current year.

File: domain/src/main/java/uk/gov/moj/sdt/domain/PurgeJobAudit.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    @Column(name = "SUCCESS")
+    private Boolean success;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+
+    }
```
Comment (paulridout) [inline domain/src/main/java/uk/gov/moj/sdt/domain/PurgeJobAudit.java:56]: Is this method intended to be empty?

File: domain/src/main/java/uk/gov/moj/sdt/domain/PurgeJobAudit.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+import java.time.LocalDateTime;
+
+/**
+ * Purge Job Audit.
+ *
+ * @author Mark Dathorne
+ */
+@Table(name = "PURGE_JOB_AUDIT")
+@Entity
+public class PurgeJobAudit implements IDomainObject {
```
Comment (paulridout) [inline domain/src/main/java/uk/gov/moj/sdt/domain/PurgeJobAudit.java:20]: Should this class extend AbstractDomainObject like the other domain objects do?

**Review by paulridout — CHANGES_REQUESTED** [review]
